src/kernel/outwards.jl

```diff
@@ -32,21 +32,37 @@ is occupied.
 
 Pass grid name `Symbol`s to `R` and `W` type parameters to use specific grids.
 """
+# Updated the OutwardsDispersal rule to include an optional mask
 struct OutwardsDispersal{R,W,S<:Stencils.AbstractKernelStencil} <: SetNeighborhoodRule{R,W}
     stencil::S
+    mask::Union{Nothing, Array{Bool}}  # Add an optional mask parameter
```

Comment:
```suggestion
    mask::M
```

The union will be _very_ slow. Instead use a type parameter `M`.

Reply:
You will need to add `M` to the type above manually, I cant from the interface here